src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
+import org.elasticsearch.common.cli.CliToolConfig;
+import org.elasticsearch.common.cli.Terminal;
 import org.elasticsearch.common.collect.Tuple;
-import org.elasticsearch.common.http.client.HttpDownloadHelper;
-import org.elasticsearch.common.io.FileSystemUtils;
-import org.elasticsearch.common.io.Streams;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.env.Environment;
-import org.elasticsearch.index.Index;
-import org.elasticsearch.node.internal.InternalSettingsPreparer;
-
-import javax.net.ssl.HttpsURLConnection;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.PrintStream;
+import org.elasticsearch.plugins.repository.*;
+
+import java.io.FileNotFoundException;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.nio.file.*;
-import java.nio.file.attribute.BasicFileAttributes;
-import java.nio.file.attribute.PosixFileAttributeView;
-import java.nio.file.attribute.PosixFileAttributes;
-import java.nio.file.attribute.PosixFilePermission;
-import java.util.*;
-
-import static org.elasticsearch.common.Strings.hasLength;
-import static org.elasticsearch.common.io.FileSystemUtils.moveFilesWithoutOverwriting;
-import static org.elasticsearch.common.settings.ImmutableSettings.Builder.EMPTY_SETTINGS;
-
-/**
- *
- */
-public class PluginManager {
-    public static final class ACTION {
-        public static final int NONE = 0;
-        public static final int INSTALL = 1;
-        public static final int REMOVE = 2;
-        public static final int LIST = 3;
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Set;
+
+import static org.elasticsearch.common.cli.CliToolConfig.Builder.cmd;
+import static org.elasticsearch.common.cli.CliToolConfig.Builder.option;
+import static org.elasticsearch.common.cli.Terminal.Verbosity.*;
+import static org.elasticsearch.plugins.repository.ChainedPluginRepository.Builder.chain;
+import static org.elasticsearch.plugins.repository.LocalPluginRepository.Builder.localRepository;
+import static org.elasticsearch.plugins.repository.PluginDescriptor.Builder.pluginDescriptor;
+import static org.elasticsearch.plugins.repository.URLPluginRepository.Builder.urlRepository;
+
+public class PluginManager extends CliTool {
+
+    private static final CliToolConfig CONFIG = CliToolConfig.config("plugin", PluginManager.class)
+            .cmds(List.CMD, Install.CMD, Uninstall.CMD)
+            .build();
+
+    public static void main(String[] args) {
+        PluginManager manager = new PluginManager();
+
+        // TODO Remove in 2.0.0
+        checkDeprecations(manager.terminal, args);
```

Comment:
Just a note here to add a new commit which removes effectively the deprecation check.